src/main/java/info/smart_tools/smartactors/utils/reflection/conversion/ConversionToConstructorWorker.java

```
@@ -0,0 +1,124 @@
+package info.smart_tools.smartactors.utils.reflection.conversion;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Type converter based on constructors.
+ * It tries to determine suitable constructor and substitute values to it.
+ * If values are not fit constructor parameter types, them will be converted too.
+ * Works only with values arranged just as in the constructor.
+ */
+public class ConversionToConstructorWorker extends ConversionWorker {
+    private static final Logger logger = LogManager.getLogger(ConversionToConstructorWorker.class);
+
+    public ConversionToConstructorWorker() {
+    }
+
+    @Override
+    public Object transform(Class<?> targetClass, Object input) throws IllegalArgumentException {
+        if (input != null) {
+            return fromValuesToObjectWithConstructor(targetClass, input);
+        }
+        return null;
+    }
+
+    // convert some values from array or map to targetClass instance through constructor only
+    private Object fromValuesToObjectWithConstructor(Class<?> targetClass, Object obj) throws IllegalArgumentException {
+
+        List<Object> values = convertToValues(obj);
+        final int valuesCount = values.size();
+
+        List<Constructor<?>> constructorList = getProperConstructors(targetClass, valuesCount);
+        for (Constructor<?> con : constructorList) {
+            Object result = null;
+            try {
```

Very strange. Why should we ignore this exceptions. Please explain? Potentially it may a reason of huge bugs in the future.